src/Package/Impl/ProjectSystem/Commands/SendToRemoteCommand.cs

```diff
@@ -52,10 +56,28 @@ internal sealed class SendToRemoteCommand : SendFileCommandBase, IAsyncCommandGr
 
             var properties = _configuredProject.Services.ExportProvider.GetExportedValue<ProjectProperties>();
             string projectDir = Path.GetDirectoryName(_configuredProject.UnconfiguredProject.FullPath);
+
+            string fileFilterString = await properties.GetFileFilterAsync();
+            Matcher matcher = new Matcher(StringComparison.InvariantCultureIgnoreCase);
+            matcher.AddIncludePatterns(fileFilterString.Split(new string[] { ";" }, StringSplitOptions.RemoveEmptyEntries));
+
+            List<string> filteredFiles = new List<string>();
+            filteredFiles.AddRange(matcher.GetMatchedFiles(nodes.GetAllFolderPaths(_configuredProject.UnconfiguredProject)));
+
+            foreach (IProjectTree node in nodes) {
+                if (node.IsFile()) {
+                    // Add any file that user specifically selected. 
+                    // This can contain a file ignored by the filter.
+                    filteredFiles.Add(node.FilePath);
+                } 
+            }
+
             string projectName = properties.GetProjectName();
             string remotePath = await properties.GetRemoteProjectPathAsync();
 
-            await SendToRemoteAsync(nodes.GetAllFilePaths(), projectDir, projectName, remotePath, CancellationToken.None);
+            if(filteredFiles.Count > 0) {
+                await SendToRemoteAsync(filteredFiles.Distinct(), projectDir, projectName, remotePath, CancellationToken.None);
```

Comment:
So there is no progress and operation is not cancellable. What if it is a huge project?